packages/vitest/src/node/config.ts

```diff
@@ -151,6 +136,46 @@ export function resolveConfig(
   if (!resolved.deps.moduleDirectories.includes('/node_modules/'))
     resolved.deps.moduleDirectories.push('/node_modules/')
 
+  resolved.server ??= {}
+  resolved.server.deps ??= {}
+
+  const deprecatedDepsOptions = ['inline', 'external', 'fallbackCJS'] as const
+  deprecatedDepsOptions.forEach((option) => {
+    if (resolved.deps[option] !== undefined) {
+      if (option === 'fallbackCJS') {
+        console.warn(c.yellow(`${c.inverse(c.yellow(' Vitest '))} "deps.${option}" is deprecated. Use "server.deps.${option}" instead`))
+      }
+      else {
+        console.warn(
+          c.yellow(
```

Comment:
Let's not hardcode `optimizer.web`. We can assume that users with jsdom/happy-dom env should use `web` - otherwise fallback to `ssr`